tests/checks/regex-import.fish

```diff
@@ -0,0 +1,36 @@
+#RUN: %fish %s
+# Tests for importing named regex groups as fish variables
+
+# Capture first match
+echo "hello world" | string match --regex -q --import -- '(?<words>[^ ]+) ?'
+printf "%s\n" $words
+# CHECK: hello
+
+# Capture multiple matches
+echo "hello world" | string match --regex -q --import --all -- '(?<words>[^ ]+) ?'
+printf "%s\n" $words
+# CHECK: hello
+# CHECK: world
+
+# Capture multiple variables
+echo "hello world" | string match -rqI -- '^(?<word1>[^ ]+) (?<word2>.*)$'
+printf "%s\n" $word1 $word2
+# CHECK: hello
+# CHECK: world
+
+# Clear variables on no match
+set foo foo
+echo "foo" | string match -rqI -- '^(?<foo>bar)$'
+echo $foo
+# CHECK:
```

Comment:
I'd expect `string match` to not set any variable when nothing was matched.
I don't know in which situations that would make a difference.